Feature/screens/signin

duswo3o opened the pull request:
## #️⃣ Issue Number

<!--- ex) #이슈번호, #이슈번호 -->
close #19 

## 📝 요약(Summary)

<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
- 로그인 모달을 HomeScreen에 추가하여 하단에서 슬라이딩으로 등장하도록 구현하였습니다.
- 기존 MyPageScreen에 있던 로그인 버튼들을 SignInButton 컴포넌트로 분리하여 재사용성을 높였습니다.
- 로그인 화면을 새로 생성하고, App.js에 로그인 페이지를 Stack Navigator에 추가하였습니다.
- axios를 설치하고, 로그인 요청 함수(requestSignin)를 구현하여 서버와 연동하였습니다.

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [x] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [x] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)

- 상단 로그인 버튼 터치 시 모달창이 활성화됩니다
![image](https://github.com/user-attachments/assets/dcdb3e39-b162-43d7-b986-6cf6f6b954cb)

- 비로그인 상태의 마이페이지 화면입니다
![image](https://github.com/user-attachments/assets/445c7fb1-d56f-4ea9-8b2a-c8b9d758e6e2)

- 로그인 페이지 입니다
![image](https://github.com/user-attachments/assets/a8139010-6ad3-4ca1-8eb0-20fe0dc20a58)


## 💬 공유사항 to 리뷰어

<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->
- axios 요청을 보내고 응답을 받는 것 까지는 성공하였지만 아직 토큰을 저장하는 로직은 고민중에 있습니다!
- 프로젝트 최상단에 config.js 파일 생성 후 자신의 컴퓨터(노트북/데스크탑)의 ip 주소를 등록해야 합니다
```
// config.js
export const API_URL = "http://<ip 주소>:8000/api/v1/";
```

## ✅ PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

duswo3o committed:
➕ Chore: install react-native-modal

react-native-modal 설치

duswo3o committed:
✨ Feat: Add modal to login button and separate SignInButton component

- HomeScreen 로그인 버튼에 모달 기능 추가
- MyPageScreen에 있던 로그인 버튼들을 SignInButton 컴포넌트로 분리하여 재사용성 개선

duswo3o committed:
✨ Feat: add signin screen

로그인 화면 생성

duswo3o committed:
✨ Feat: add login button and navigation stack for login page

- 로그인 버튼 추가
- App.js에 로그인 페이지 Stack.Navigator 추가

duswo3o committed:
➕ Chore: install axios

api 요청을 위한 axios 설치

duswo3o committed:
✨ Feat: implement login request function

로그인 요청 함수 생성

Drills-hub reviewed (commented):
![KakaoTalk_20250612_221902181_01](https://github.com/user-attachments/assets/42f77736-cea3-48f9-a601-79700a17bd13)
![KakaoTalk_20250612_221902181](https://github.com/user-attachments/assets/89cf1225-62c5-4dd6-8a57-a4e846135736)

안드로이드 환경에서 확인해 봤는데요
회원가입이나 나이 입력 시 키보드가 올라올때 '다음'버튼이 겹쳐서 입력 부분이 안보이는 곳이 있고, 생년월일은 년도 입력시 월, 일 나이 까지 하나로 입력 되는 현상이 있습니다!

duswo3o commented:
@Drills-hub 
회원가입 부분은 백엔드에서 모델 수정 후에 추가적으로 수정예정입니다~ 로그인 부분만 봐주시면 될거같아요!

Drills-hub reviewed (approved):
로그인은 다음 버튼이 가리는 거 빼고는 좋습니다!

duswo3o committed:
🩹 Fix: adjust keyboardVerticalOffset for android

Android에서 버튼이 textinput을 가지는 문제 해결
- ScrollView로 수정
- keyboardVerticalOffset keyboardShouldPersistTaps 설정
